compiler/rustc_resolve/src/rustdoc.rs

```diff
@@ -350,10 +350,7 @@ pub fn strip_generics_from_path(path_str: &str) -> Result<Box<str>, MalformedGen
 /// If there are no doc-comments, return true.
 /// FIXME(#78591): Support both inner and outer attributes on the same item.
 pub fn inner_docs(attrs: &[impl AttributeExt]) -> bool {
-    attrs
-        .iter()
-        .find(|a| a.doc_str().is_some())
-        .map_or(true, |a| a.style() == ast::AttrStyle::Inner)
+    attrs.iter().find(|a| a.doc_str().is_some()).is_none_or(|a| a.style() == ast::AttrStyle::Inner)
```

Comment:
the logic here could be simplified. is this not just `attrs.iter().any(|a| a.doc_str().is_some() && a.style() == ast::AttrStyle::Inner)`?

Reply:
Err.. I don't think so, the original logic returns true if `find()` returns `None`. No?
Can't wrap my head right now around what's happening here, but I definitely agree it's not very readable.

Reply:
Oh, right. What the heck.